disnake/subscription.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+import datetime
+from typing import TYPE_CHECKING, List, Optional
+
+from .enums import SubscriptionStatus, try_enum
+from .mixins import Hashable
+from .utils import parse_time, snowflake_time
+
+if TYPE_CHECKING:
+    from .state import ConnectionState
+    from .types.subscription import Subscription as SubscriptionPayload
+    from .user import User
+
+__all__ = ("Subscription",)
+
+
+class Subscription(Hashable):
+    """Represents a subscription.
+
+    This can only be retrieved using :meth:`SKU.subscriptions` or :meth:`SKU.fetch_subscription`.
```

Comment:
```suggestion
    This can only be retrieved using :meth:`SKU.subscriptions` or :meth:`SKU.fetch_subscription`,
    or provided by events (e.g. :func:`on_subscription_create`).
```